.github/workflows/linting.yml

```diff
@@ -10,12 +10,12 @@ concurrency:
 
 jobs:
   python-linting:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     steps:
-    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # 4.2.2
+    - uses: actions/checkout@v4
```

Comment:
Using a specific commit is deliberate, out of abundance of care.

We'll only update to newer versions/commits when really needed, so please change back.